### ISSUE TITLE
Chromecast not detected - network segments

### ISSUE THREAD

**Issue by 150d** [angelnu/ioBroker.chromecast#41]
Hi,

the adapter does not detect my Chromecast. The reason is probably that the two (ioBroker and Chromecast device) reside on seperate network segments, so discovery by SSDP/MDNS cannot be reliable.

But why is there no manual way to set the IP of the chromecast device?

Regards

**Comment by angelnu** [comment]
Adding manually selected devices would be possible. It would be mostly an extension of the admin UI to set up their parameters. These are the parameters used by the adapter that comes from the DDNS record:
```
      this.id = id;
      this.name = name;
      this.host = host;
      this.port = port;
      this.type = type; //'Chromecast','Chromecast Audio','Google Cast Group'
```

Btw: at my home I am also running iobroker into a different network as the Chromecast devices. I added a DDNS repeater in a node that has access to both networks. If this is an option for you here is the link to the [Kuberntes recipe](https://github.com/angelnu/docker-mdns_repeater/blob/master/kubernetes.yaml) I use

**Comment by 150d** [comment]
I'm afraid I'm not able to extend the adapter myself, even with your information. I tried to look at the code but don't even understand where the search happens or where new devices are discovered.

**Comment by angelnu** [comment]
I can do the part that handles the chromecast instances but my UI admin skills are very limited.

It would need a table like this:
![image](https://user-images.githubusercontent.com/4406403/50742061-8d39f400-1206-11e9-9892-b8fe9b1ead0c.png)
very similar to the google geofence adapter. The "+" should allow to add rows with manual values for *name*, *ip*, *port* and *type*.

I have created a branch for this feature at https://github.com/angelnu/ioBroker.chromecast/blob/manualDevices/admin/index_m.html#L173

But I have no idea of how to make the "+" button to work like it does here: https://github.com/t4qjXH8N/ioBroker.google-sharedlocations/blob/master/admin/index_m.html#L308

I am stuck until someone helps with the UI.

**Comment by 150d** [comment]
Can the adapter handle multiple Chromecast devices in a single instance? If not I would propose to just add input boxes for a single device, without the dynamic "+" (which would allow any number of devices). Do you think that would be enough? A second device could then be handled in a second instance of the adapter.

**Comment by angelnu** [comment]
Yes, the adapter supports multiple devices with a single instance. This was very important for me since I run an instance with 15 chromecast devices with only 80 MB memory consumption.

What I can do for the "short" term until someone helps with the UI is similar to your idea: have a a field where a json structure can be pasted. Let me see if I can add it for you to test.

**Comment by angelnu** [comment]
Ok, I figured it with the table in the admin view and created a first version with support for adding new devices. Please update to https://github.com/angelnu/ioBroker.chromecast/tarball/manualDevices

It is a very first dev pre (2.2.0) without any sanity check for the parameters so please ensure that no special characters are used for the name. I will add the checks during the week and then release to npm.

**Comment by angelnu** [comment]
I merged the code into master and npm 2.2.0.

**Comment by 150d** [comment]
Working great, thank you!
